### PR TITLE
DPL: Fixing one-site gap checks for multi-height cells

### PR DESCRIPTION
**Pull request by mina1460**
Here is the updated test case:
Screenshot 2023-06-14 at 11 41 52 pm

Running the modified check_placement_cmd 
```
check_placement -disallow_one_site_gaps -verbose
```

Output:
```
[WARNING DPL-0007] One site gap check failed (3).
 r1
 t1
 d1
```

Now, we run 
` >>> detailed_placement -disallow_one_site_gaps `
The output is:
```
Placement Analysis
---------------------------------
total displacement          0.2 u
average displacement        0.1 u
max displacement            0.2 u
original HPWL               0.0 u
legalized HPWL              0.0 u
delta HPWL                    0 %

```

and we can see in the GUI that the gap was detected and eliminated

Screenshot 2023-06-14 at 11 42 46 pm

### CODE REVIEW

**Comment by maliberty** [comment]
ready for review?

**Comment by mina1460** [comment]
@maliberty not yet, still fixing some small issues

File: src/dpl/src/Grid.cpp [hunk 1]
```diff
@@ -329,7 +329,10 @@ void Opendp::visitCellBoundaryPixels(
       int x_end = gridEndX(rect.xMax() - core_.xMin(), site_width);
       int y_start = gridY(rect.yMin() - core_.yMin(), row_height);
       int y_end = gridEndY(rect.yMax() - core_.yMin(), row_height);
-
+      // TODO: This is used in the check one site gaps logic, but it doesn't do
```
Comment (maliberty) [inline src/dpl/src/Grid.cpp:332]: Does this need to be resolved?

**Review by maliberty — APPROVED** [review]

**Comment by maliberty** [comment]
Holding the merge pending clang-tidy cleanup

**Pull request merged**